DuplicateFinderEngine/FileHelper.cs

```diff
@@ -1,81 +1,32 @@
+		public static readonly string[] AllExtensions = VideoExtensions.Concat(ImageExtensions).ToArray();
 
 		// '' <summary>
 		// '' This method starts at the specified directory.
 		// '' It traverses all subdirectories.
 		// '' It returns a List of those directories.
 		// '' </summary>
 		public static List<string> GetFilesRecursive(string initial, bool ignoreReadonly, bool recursive, bool includeImages, List<string> excludeFolders) {
-			var result = new List<string>();
-			var stack = new System.Collections.Concurrent.ConcurrentStack<string>();
-			stack.Push(initial);
-			while (stack.Count > 0) {
-				stack.TryPop(out var dir);
-				try {
-					var skip = false;
-					var DirInfo = new DirectoryInfo(dir);
-
-					if (DirInfo.Attributes.HasFlag(FileAttributes.ReadOnly) && ignoreReadonly) {
-						Logger.Instance.Info(string.Format(Properties.Resources.SkippedReadonly, dir));
-						skip = true;
-					}
-
-					for (var i = 0; i < excludeFolders.Count; i++) {
-						if (!dir.Contains(excludeFolders[i])) continue;
-						Logger.Instance.Info(string.Format(Properties.Resources.SkippedBlacklist, dir));
-						skip = true;
-						break;
-					}
-
-					if (!skip) {
-						result.AddRange(Directory.GetFiles(dir, "*.avi"));
-						result.AddRange(Directory.GetFiles(dir, "*.mkv"));
-						result.AddRange(Directory.GetFiles(dir, "*.flv"));
-						result.AddRange(Directory.GetFiles(dir, "*.mov"));
-						result.AddRange(Directory.GetFiles(dir, "*.mpg"));
-						result.AddRange(Directory.GetFiles(dir, "*.mpeg"));
-						result.AddRange(Directory.GetFiles(dir, "*.wmv"));
-						result.AddRange(Directory.GetFiles(dir, "*.mp4"));
-						result.AddRange(Directory.GetFiles(dir, "*.m4v"));
-						result.AddRange(Directory.GetFiles(dir, "*.asf"));
-						result.AddRange(Directory.GetFiles(dir, "*.f4v"));
-						result.AddRange(Directory.GetFiles(dir, "*.webm"));
-						result.AddRange(Directory.GetFiles(dir, "*.divx"));
-						result.AddRange(Directory.GetFiles(dir, "*.m2t"));
-						result.AddRange(Directory.GetFiles(dir, "*.m2ts"));
-						result.AddRange(Directory.GetFiles(dir, "*.vob"));
-						if (includeImages)
-							foreach (var s in ImageExtensions)
-								result.AddRange(Directory.GetFiles(dir, $"*{s}"));
-					}
-
-					if (!skip && recursive) {
-						Parallel.ForEach(
-						   Directory.GetDirectories(dir), directoryname => stack.Push(directoryname)
-						   );
-					}
-
-				}
-				catch (Exception ex) {
-					// ReSharper disable once RedundantStringFormatCall
-					System.Diagnostics.Trace.TraceError(string.Format(Properties.Resources.SkippedErrorReason, dir, ex.Message));
-				}
-
-			}
-
-			return result;
+			var files = Directory.EnumerateFiles(initial).Where(f => (includeImages ? AllExtensions : VideoExtensions).Any(x => f.EndsWith(x, StringComparison.OrdinalIgnoreCase)));
```

Comment:
Wouldn't that break enumeration on access violations. Did you tested this against for example a drive root such as C:\ where it won't have permission to folders like $recycle.bin ?

Reply:
yep, let me make this a bit more robust

Reply:
fixed. btw are you using read-only folders? it seems the only way to make a folder (not the files inside) read only, is to use the command line. e.g.: "attrib +r d:/test_folder"

Reply:
I dont but who knows what other people do on their system and what other software did. Regarding file system I prefer to be safe.